src/legacy/core_plugins/timelion/public/legacy.ts

```diff
@@ -21,13 +21,12 @@ import { PluginInitializerContext } from 'kibana/public';
 import { npSetup, npStart } from 'ui/new_platform';
 import { plugin } from '.';
 import { setup as visualizations } from '../../visualizations/public/np_ready/public/legacy';
-import { setup as data } from '../../data/public/legacy';
 import { TimelionPluginSetupDependencies } from './plugin';
 import { LegacyDependenciesPlugin } from './shim';
 
 const setupPlugins: Readonly<TimelionPluginSetupDependencies> = {
   visualizations,
-  data,
+  data: npSetup.plugins.data,
```

Comment:
If `timefilter` was re-exported in the shim, this change would not be necessary. Also, now if Timelion needs some other service besides `timefilter` from the `data` plugin they would need to import the shim and the NP plugin, merge those locally, also import types of both of those and merge those locally.